packages/next/server/dev/on-demand-entry-handler.ts

```diff
@@ -545,9 +556,9 @@ export function onDemandEntryHandler({
       try {
         const pagePathData = await findPagePathData(
           rootDir,
-          pagesDir,
           page,
           nextConfig.pageExtensions,
+          pagesDir,
```

Comment:
It looks like the 404 page is stalling in development when the pages dir isn't present, possibly we aren't ensuring correctly without the directory detected

Reply:
Nice catch, fixed and added a test for it